gwcs/coordinate_frames.py

```diff
@@ -396,11 +296,15 @@ def __init__(self, frames, name=None):
         for frame in frames:
             axes_order.extend(frame.axes_order)
         for frame in frames:
-            for ind, type, u, n in zip(frame.axes_order, frame.axes_type,
-                                          frame.unit, frame.axes_names):
+            for ind, type, un, n in zip(frame.axes_order, frame.axes_type,
+                                        frame.unit, frame.axes_names):
```

Comment:
While you are modifying things in here, you might want to change "type" here to some other variable name as "type" is a common function and could cause odd collisions.  Maybe "axtype"